### PR TITLE
Ability to proceed runInTerminal request in sidecar container

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### Reference issue
https://github.com/eclipse/che/issues/12771

Method overriding is implemented in https://github.com/eclipse/che-theia/pull/107

### CODE REVIEW

File: packages/debug/src/browser/debug-session.tsx [hunk 1]
```diff
@@ -63,7 +64,7 @@ export class DebugSession implements CompositeTreeElement {
 
     constructor(
         readonly id: string,
-        readonly options: DebugSessionOptions,
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-session.tsx]: that's an API, would be nice to preserve it as `options`

File: packages/plugin-ext/src/api/plugin-api.ts [hunk 2]
```diff
@@ -978,6 +978,7 @@ export interface DebugExt {
     $getConfigurationSnippets(debugType: string): Promise<IJSONSchemaSnippet[]>;
     $createDebugSession(debugConfiguration: theia.DebugConfiguration): Promise<string>;
     $terminateDebugSession(sessionId: string): Promise<void>;
+    $getTerminalCreationOptions(debugType: string): Promise<any>;
```
Comment (akosyakov) [inline packages/plugin-ext/src/api/plugin-api.ts]: Could be here better type than `any`, i.e.  `TerminalOptionsExt | undefined`?

**Comment by akosyakov** [comment]
How one will be able to test it?

**Comment by tolusha** [comment]
@akosyakov 
* applied  https://github.com/eclipse/che-theia/pull/107
* built custom images for  `che-theia`, `che-theia-endpoint-runtime`, `che-remote-plugin-python-3.7.2` 
* created a workspace in a minikube with Python extension
* started a debug session

Debug process was run in a sidecar container
```
PID   USER     TIME  COMMAND
    1 root      0:02 node /home/theia/lib/node/plugin-remote.js
   48 root      0:00 /usr/local/bin/python completion.py
   51 root      0:00 /usr/local/bin/python lib/python/jedi/evaluate/compiled/subprocess/__main__.py lib/python
  129 root      0:00 node /plugins/ms-python.python.2019.2.5433/extension/out/client/debugger/debugAdapter/main.js
  139 root      0:00 /usr/local/bin/python /plugins/ms-python.python.2019.2.5433/extension/pythonFiles/ptvsd_launche
  163 root      0:00 /bin/ash
  169 root      0:00 ps
```

**Pull request merged**